src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    static final ColorSupplier colorSupplier = new ColorSupplier();
+    static final int maxParameterValue = 50;
+    static final int FIGURES_COUNT = 5;
+    static final Random random = new Random();
+    static final int DEFAULT_RADIUS = 10;
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        double side = new Random().nextInt(maxParameterValue);
+        double width = new Random().nextInt(maxParameterValue);
+        double height = new Random().nextInt(maxParameterValue);
+        double radius = new Random().nextInt(maxParameterValue);
+        double base = new Random().nextInt(maxParameterValue);
+        double firstLeg = new Random().nextInt(maxParameterValue);
+        double secondLeg = new Random().nextInt(maxParameterValue);
+        int value = random.nextInt(FIGURES_COUNT);
+        switch (value) {
+            case 0:
+                return new Square(color, side);

```

Comment:
Good! But do we actually need to declare every variable every time? If we create a Squre - do we need a radius created?

```suggestion
        String color = colorSupplier.getRandomColor();
        double width = new Random().nextInt(maxParameterValue);
        double height = new Random().nextInt(maxParameterValue);
        double radius = new Random().nextInt(maxParameterValue);
        double base = new Random().nextInt(maxParameterValue);
        double firstLeg = new Random().nextInt(maxParameterValue);
        double secondLeg = new Random().nextInt(maxParameterValue);
        int value = random.nextInt(FIGURES_COUNT);
        switch (value) {
            case 0:
                double side = random.nextInt(maxParameterValue);
                return new Square(color, side);

```

And so on for every other case.